common/multistep_runner.go

```diff
@@ -0,0 +1,163 @@
+package common
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/packer"
+)
+
+func newRunner(steps []multistep.Step, config PackerConfig, ui packer.Ui) (multistep.Runner, multistep.DebugPauseFn) {
+	switch config.PackerOnError {
+	case "cleanup", "":
+	case "abort":
+		for i, step := range steps {
+			steps[i] = abortStep{step, ui}
+		}
+	case "ask":
+		for i, step := range steps {
+			steps[i] = askStep{step, ui}
+		}
+	default:
+		ui.Error(fmt.Sprintf("Ignoring unknown on-error value %q", config.PackerOnError))
```

Comment:
Consider adding the name of the configuration value to make the error message clearer for end-users if they provide a bad value.